src/mesh/interpolation.cxx

```diff
@@ -44,12 +44,16 @@ BoutReal interp(const stencil &s)
 /*!
   Interpolate between different cell locations
   
-  NOTE: This requires communication
+  NOTE: This requires communication if the result is required in guard cells
+  NOTE: Since corner guard cells cannot be communicated, it never makes sense
+  to calculate interpolation in guard cells. If guard cell values are required,
+  we must communicate (unless interpolating in z).
 
   @param[in]   var  Input variable
   @param[in]   loc  Location of output values
+  @param[in]   region  Region where output will be calculated
 */
-const Field3D interp_to(const Field3D &var, CELL_LOC loc)
+const Field3D interp_to(const Field3D &var, CELL_LOC loc, REGION region)
```

Comment:
The region argument is ignored in most cases which might be a bit unexpected. A comment to note this may help.

Reply:
Agreed, I'll add something tomorrow.

The cases except RGN_NOBNDRY are all basically the same because we always need to communicate to get valid guard cells: the corner guard cells aren't ever set, so we can't calculate the interpolation in guard cells. Also, the communicate() goes in both x- and y-directions, so needs to be called for any of nox/noy/all.
If there were 1d communication routines, then e.g. RGN_NOX could communicate just y-guard cells. Maybe that would offer a small speedup?